# Notes while editing on 2017-Aug-08 by Jesse Bloom
* First paragraph, first section of Results: 
    - First sentence: I have simplified *"In order to understand influenza replication dynamics we measured the transcriptome of individual cells using a droplet-based system -- physically isolating cells prior to performing reverse-transcription."* to just be a simpler sentence that says what we did directly. The current sentence is too complex. I think the end of the Introduction will probably introduce the concept of what we are doing and why, so we probably don't need to repeat that here.
    - Last sentence: This sentence was also a bit complex, so I've broken it into two sentences each with one thought.

* Second paragraph, first section of Results:
    - It seemed important to mention the viral strain somewhere early, so I added that.
    - Re-worded to indicate that co-infected cells with **often** express mRNAs from both virus barcodes. It won't happen if they get co-infected with the same barcode.

* Figure 1 legend:
    - I have shortened the first sentence to just say *"Experimental overview"* because I think the other title was too complex to be useful. I've also tried to clarify other parts.
    - I have removed the part about the mRNA numbering being in mRNA coordinates in panel A. I do think this is an important point, but it's too subtle and complex to make here. In fact, I think it would be better to **remove the mRNA sequencing numbering from panel A altogether.** It's not really important for conceptually understanding the idea, , and we give the full vRNA sequences in the figure source data.
    - I created the source data with the vRNA sequences.

* Third paragraph, first section of Results:
    - My main goal was to shorten this paragraph, which was too long for a point this technical.
    - I have changed *"diversify into an array of biologically active particles"* to just indicate that the stocks typically are an array. In our case, it's not clear that the number of this particles are increasing due to ongoing diversification versus just being there the whole time.
    - I've gotten rid of the part about the gene absence and deletions in the polymerase. This is too much detail for this point in the text, where the only point that we want to make is that some viruses are defective. We can come back to these details if they become relevant later
    - I have re-written the text to say *higher ratio of infectious particles to RNA* because it is more intuitive way to relate to relate *higher purity* to *higher ratio* than to *lower ratio*. I know this makes the statement refer the inverse of what is plotted in Figure 2. An easy fix would be to invert axis in Figure 2 by plotting the TCID50 / HA instead.

* Third section of results: 
    - The term *"non-productive"* is too general / vague to use for segment absence, since lots of factors can make a viral infection non-productive in terms of progeny virions. I've therefore tried to be more specific by saying that we are looking for gene absence, and avoid statements about whether or not the infection is productive since we never measure progeny virus production.
    - The rest of this section seems quite good to me. However, if we want the paper to be readable by non-virologists, we need to motivate the concept of the segmented genome and the RNP before jumping into the results, so I've re-structured a bit to do that.
    - I moved reference to the cyclohexamide figure earlier to support the point that secondary transcription is important.

* Other sections:
    - I stopped detailed editing at the fourth section (*Influenza mRNA Ratios are Tightly Controlled*), since that appeared to be where you had stopped making lots of changes. I did move the last section (*Relative expression of different viral genes*) up to be part of that section, since the seem the same.
    - When you're writing this section, my main comment would be that the current text gets the main points but might be making the claims a bit too strongly given the supporting evidence.
