%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ELIFE ARTICLE TEMPLATE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% PREAMBLE 
\documentclass[9pt,lineno]{elife}
% Use the onehalfspacing option for 1.5 line spacing
% Use the doublespacing option for 2.0 line spacing
% Please note that these options may affect formatting.

\usepackage{lipsum} % Required to insert dummy text
\usepackage[version=4]{mhchem}
\usepackage{siunitx}
\DeclareSIUnit\Molar{M}
\usepackage{indentfirst} %I prefer to have indents following section headings, I find it to be more consistent in style throughout document

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE SETUP
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Transcriptional dynamics of influenza virus infection at the single-cell level}

\author[1]{Alistair B. Russell}
\author[2]{Cole Trapnell}
\author[1,2*]{Jesse D. Bloom}
\affil[1]{Basic Sciences Division and Computational Biology Program, Fred Hutchinson Cancer Research Center, Seattle, United States}
\affil[2]{Department of Genome Sciences, University of Washington, Seattle, United States}
\corr{jbloom@fredhutch.org}{}

% \presentadd[\authfn{5}]{eLife Sciences editorial Office, eLife Sciences, Cambridge, United Kingdom}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE START
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\maketitle

\begin{abstract}
Influenza virus infection induces large changes in cellular transcription.
Previously this has mostly been looked at using bulk measurements
Here we examine the process at the level of single cells.
We find extremely wide variation in the extent of viral gene transcription across infected cells.
IFN induction is very rare.
Some cellular pathways may be consistently altered in cells with high burden of viral transcripts.
Overall, highlights remarkable heterogeneity in the outcome of infection.
\end{abstract}


\section{Introduction}

Heterogeneity is important in a lot of cellular processes even when isogenic~\citep{shalek2013single,shalek2014single}.

Population (genetic) heterogeneity is also important. 
Viral quasispecies, cancer single-cell, etc.
Salmonella paper (PhoP).

Literature on viral burst-size heterogeneity.
This goes back to Delbruck, Andino polio paper~\citep{schulte2014single}, the MDCK / flu paper.

Discuss segmented nature of influenza.
Maybe in the context of how this could further increase heterogeneity because there is a lot of potential for entire genes to be missing.
Includes Yewdell and Lowen papers.

\section{Results}

\subsection{Initial Steps to Generate Robust Single-Cell Kinetic Observations.}


	In order to obtain detailed measurements at a single-cell resolution of influenza growth and replication we chose to use an approach dependent on a commercially available variant of DropSeq and InSeq (Figure~\ref{fig:workflow}A). In brief, we infected A549 cells --- a human lung carcinoma cell line frequently used in tissue-culture models of influenza infection --- using a relatively low infectious dose of WSN influenza --- a lab-adapted trypsin-independent strain. We then harvested cells by trypsonization, and generated libraries from total polyadenylated transcripts. These libraries consisted of a short region mapping to the 3? end of the original mRNA transcript, a barcode identifying the cell from which the transcript was derived, and a unique molecular identifier that serves for error-correction in transcript abundance measurements. As an additional control, we generated a 3? barcoded influenza strain containing two one base-pair synonymous mutations at the 3? end of all transcripts, allowing us to estimate co-infection and empirically derive thresholds for influenza infection. These barcodes were designed to lie within the 3? region sequenced in our library preparation as determined through examination of mapping distributions in uninfected A549 cells.
	
        	When generating viral stocks for our infections, we also considered that Influenza, like many viruses, can generate a diverse array of biologically active particles as it replicates --- particles that can differ significantly from the genomic consensus sequence. One of the most abundant of these, defective-interfering (DI) particles, predominantly result from deletions accrued in the three segments encoding the tripartite polymerase during growth at high infectious doses. Generation of these particles is highly stochastic and known to influence the behavior of influenza populations both with respect to replication kinetics and interferon induction. Therefore, for the sake of generating baseline, repeatable, observations of influenza replication kinetics and host interactions at a single cell level, we first sought to propagate populations of viruses that were relatively free of these particles using similar methods to those developed by \textit{Xue et al}. Concomitantly, we validated that both replication kinetics and relative abundance of DI particles were similar between our synonymously barcoded and wild-type populations. 
	
	First, we established that our barcoded variant progresses through infection with near-wild-type kinetics, as measured by hemagglutanin (HA) transcript abundance (Figure~\ref{fig:viruspopulations}A). After establishing that our mutations did not dramatically impact viral replication, we next sought to establish a rough ratio of biologically active particles to fully-infectious particles.  We therefore measured the abundance of the HA segment in viral stocks relative to their measured infectious dose (Figure~\ref{fig:viruspopulations}B). Notably, this ratio is dramatically skewed in a population grown to be replete in defective particles relative to our desired experimental populations. Measurements of HA staining corroborated this finding, as the percentage of infected cells tracked well with the desired MOI, as opposed to a dramatically increased abundance of HA staining noted in our high defective control  (Figure~\ref{fig:viruspopulations}C). Lastly, our populations induced far less interferon than a high defective control, consistent with prior observations that such stocks are more immunostimulatory (Figure~\ref{fig:viruspopulations}D).
	
	Using these validated populations, we tracked growth kinetics in our target cell line at an infectious dose targeting a final infection rate of 5 percent as determined by flow cytometry. We observed rapid transcript accumulation through 10 hours, which then tapered off --- potentially representing new rounds of infection rather than progression in initially infected cells (supplemental figure). We then chose three time points (6,8, and 10h) reflecting dramatically different influenza burden (and thus different stages in influenza replication) for this study.



 (Figure~\ref{fig:workflow}).

\begin{figure}
\includegraphics[width=0.8\linewidth]{figures/Workflow/workflow.pdf}
\caption{
CAPTION}
\label{fig:workflow}
\end{figure}

\begin{figure}
\includegraphics[width=0.7\linewidth]{figures/Validating_barcode_virus/validating_populations_D02.pdf}
\caption{
CAPTION}
\label{fig:viruspopulations}
\end{figure}

\subsection{Population-Level Measurements Validate Dataset.}
Before proceeding with more detailed analyses, we first examined population-level data to confirm the validity of our approach. We not only generated data for our three time points and a matched uninfected control, but also performed a biological replicate at 8 hours to determine whether our findings could be dominated by inter-sample or technical variation rather than biological processes driven by influenza. We sampled roughly equivalent number of cells per time point at an equivalent sequencing depth per cell - granting similar probability to observe rare events across datasets. Overall we observed highly similar distributions of transcript abundance between samples. This suggests that the cellular state is on average relatively unperturbed and that influenza, if it is driving dramatic changes in transcript abundance and/or growth state, must be doing so in only a small subset of cells. More importantly, with the confidence that the average cellular state is similar between time points, we can better regard changes in subpopulations between time points as bone-fide progression in influenza replication rather than inter-sample variation.

In addition to analyses regarding global transcript abundance, we were also able to successfully detect increasing numbers of influenza transcripts as our experiment progressed. Moreover, there appeared to be only a few cells at each time point making considerable contributions to the influenza transcript pool, consistent with our desired low level of infection. With these metrics, we felt confident proceeding to more detailed analyses of influenza replication kinetics and impact on these populations.

We got some reasonable number of cells at several timepoints (Figure~\ref{fig:cells}A).

Most cells are uninfected, flu burden generally increases with time in the samples, show Figure~\ref{fig:cells}B maybe with inset.
Also show Figure~\ref{fig:cells}C which basically shows the same thing in a different way.

We observe both barcodes in influenza-infected cells, and can determine the barcode for about half the viral transcripts (Figure~\ref{fig:synbarcodes}).

\begin{figure}
\includegraphics[width=\linewidth]{figures/p_cell_mRNA_summary.pdf}
\caption{
Overview of amounts of cellular and influenza virus mRNAs detected in each cell.
{\bf (A)} 
Number of cells captured for each sample.
{\bf (B)} 
Cumulative fraction plot showing the amount of mRNA derived from influenza for each sample.
In all samples, most cells had little or no influenza mRNA.
{\bf (C)} 
The number of cellular and viral mRNAs for each cell is plotted as a point.
The blue lines show the overall distribution of the number of cellular mRNAs per sample.
Cells that fell outside the red lines were removed as outliers.
At later timepoints, a small number of cells had a very high number of viral mRNAs.
}
\label{fig:cells}
\figsupp[Shorter caption for main text.]{This is a supplementary figure's full caption, which will be used at the end of the manuscript.}{\includegraphics[width=6cm]{frog}}
\figsupp{This is another supplementary figure.}{\includegraphics[width=6cm]{frog}}
\figdata{This is a description of a data source.}
\end{figure}

\begin{figure}
\includegraphics[width=\linewidth]{figures/synbarcodes_umistats.jpg}
\caption{
CAPTION}
\label{fig:synbarcodes}
\end{figure}

\subsection{Barcodes enable systematic identification of infected and co-infected cells.}
Few coinfections

Setting threshold: Assumes rare barcode events are due to lysis (ormisannotation)  and that two values SHOULD correlate
Validating evidence/assumption two values correlate - flow data, flow diagram and quantile-normalized data. Include replicates in supplement.

Once defined, we have some statistics.

Presumably this subsection basically covers \ref{fig:fracflu}.

\begin{figure}
\includegraphics[width=\linewidth]{figures/p_frac_flu_summary.pdf}
\caption{
Synonymous barcodes near the 3' end of the influenza virus mRNAs were used to identify co-infection and distinguish true infections from cells that contained a few spurious viral reads.
{\bf (A)}
For all cells with at least two viral mRNAs for which the synonymous barcode could be called, each line is proportional to the logarithm of the number of viral mRNAs in that cell.
The bars are colored in linear proportion to the fraction of the viral mRNAs derived from either wildtype or synonymously barcoded virus.
{\bf (B)}
Same as (A), but now each bar is colored according to the relative proportions of the more common (major) and less common (minor) barcoded virus variant.
At low levels of viral mRNA there is often a roughly equal mix of barcodes, since many of these cells have simply picked up environment mRNA which is about equally likely to derive from either virus.
But at higher levels of viral mRNA, truly infected cells are mostly one pure barcode except for a few cells that are truly co-infected.
{\bf (C)}
We determined a cutoff for calling ``true'' infections by fitting a curve to the mean barcode purity of all cells with greater than a given fraction of their mRNA derived from virus.
We called the cutoff at the point at which purity stops increasing with the fraction of viral mRNA.
{\bf (D)}
The number of cells identified as infected and co-infected for each sample.
For all samples, the vast majority of cells were not infected, so for subsequent analyses we subsampled to a number of uninfected cells that was the greater of 50 or the number of infected cells.
{\bf (E)} 
The distribution of the fraction of mRNA derived from virus for each sample for both infected and co-infected cells.
For all samples, there is a very wide distribution of the amount of viral mRNA.
}
\label{fig:fracflu}
\end{figure}

\subsection{Absence of viral genes explains some of the heterogeneity in viral burden.}
In the prior sections, we've pointed that there is a lot of variation in how much flu there is in each cell.
One obvious explanation is some cells essential flu genes.
Most obviously, viruses missing a polymerase gene are presumably have problems making transcripts.
Possibly one could imagine that missing other genes would have an effect.

Figure~\ref{fig:fluburdenbygene} shows that viruses missing polymerase genes never get up as high in flu burden.
Also work in as a figure supplement the qPCR of the cycloheximide.

Talk about caveats of calling polymerase absence.

Potentially add a figure or figure panel showing that there is still a lot of heterogeneity in 8-segment virus.

\begin{figure}
\includegraphics[width=\linewidth]{figures/p_flu_burden_flu_gene_merge.pdf}
\caption{
The viral infection burden in individual cells as a function of the amount of each viral gene detected.
{\bf (A)} 
Fraction of mRNAs in each cell derived from virus as a function of the \emph{normalized} expression of each viral gene in that cell.
This plot shows that all cells with very high viral burden express all of the RNP genes, but some cells with high viral burden lack each of the other four viral genes.
{\bf (B)}
Statistical tests confirming that absence of viral RNP genes is significantly associated with reduced viral burden, but that the absence of the non-RNP genes does not lead to a clear decrease in viral burden.
}
\label{fig:fluburdenbyflugene}
\end{figure}

\subsection{How many cells are missing influenza genes?}
Look at the non-polymerase genes in highly infected cells, and see how often they are absent.
This probably requires a new figure.

Talk about how co-infection can rescue absence of genes.
From our single-cell data, Figure~\ref{fig:coexpression} shows some of the relevant data.

From the other data, talk about the flow cytometry data.
Coinfection gives multiple 'chances' bring up flow data (hiDI and co-infection, as well as available data, bring up prior Yewdell data consistent with this).

\begin{figure}
\includegraphics[width=\linewidth]{figures/p_coinfection.pdf}
\caption{
Frequency of each viral gene segment in co-infected cells with at least 5\% of their mRNA derived from influenza.
The bars indicate the logarithm of the number of each viral mRNA detected, and the bars are colored in proportion to the fraction of those mRNAs that are derived from either wildtype or synonymous barcoded virus.
}
\label{fig:coexpression}
\figdata{The raw data plotted in this figure are in \texttt{p\_coinfection.csv}.}
\end{figure}

\subsection{Relative expression of different viral genes}
Figure~\ref{fig:fluexpr} shows some of this.
Cite \citet{hatada1989control} to show our relative expressions are consistent with that.
Maybe show a panel with just the 8-segment infections.

Maybe add a figure breaking this down among timepoints.

\begin{figure}
\includegraphics[width=\linewidth]{figures/p_flu_expr.pdf}
\caption{
Expression of individual influenza genes in highly infected cells (at least 5\% of total mRNA is viral).
{\bf (A)} 
The fraction of total mRNA from each influenza gene for each cell.
{\bf (B)}
Box plots showing the fraction of viral mRNA per cell that is derived from each influenza gene taken over all highly expressed cells.
The black line at the notch in each box is the median, and the top and bottom of the box indicate the first and third quartiles.
}
\label{fig:fluexpr}
\figdata{The raw data are in \texttt{p\_flu\_expr.csv}.}
\end{figure}

\subsection{Host stuff}

\begin{figure}
\includegraphics[width=\linewidth]{figures/p_pseudo_merge.pdf}
\caption{
Visual layout of the cells according to ``pseudotime''.
The layout is the same in all panels, but each panel colors the cells according to a different property.
{\bf (A), (B)}
Cells colored by the fraction of their mRNA that is viral.
{\bf (C)}
Cells colored by experimental sample.
While it is clear that cells from later timepoints often have more viral RNA, there are cells from earlier timepoints with a high viral burden and cells from late timepoints with a low viral burden.
{\bf (D)}
Cells colored by the number of type I and III interferon transcripts detected.
Only one cell has high expression of these interferons.
{\bf (E)}
Cells colored by the expression of the interferon-stimulate gene ISG15.
{\bf (F)}
For cells with at least 1\% of their reads from influenza, are the cells expressing the viral NS protein?
The one interferon-positive cell is lacking NS, but many other cells also lack NS but do not express interferon.
}
\label{fig:pseudotime}
\end{figure}


\begin{figure}
\includegraphics[width=0.8\linewidth]{figures/p_cellular_heatmap.pdf}
\caption{
Cellular genes that are differentially expressed with respect to the amount of influenza mRNA in individual cells infected with full influenza virus containing all eight genes.
Shown are all genes differentially expressed with $Q < 0.1$.}
\label{fig:cellulargenes}
\figdata{The full results of the differential expression test is in \texttt{p\_sig\_cellular\_genes.csv}.}
\figdata{The results of a gene-set analysis are in \texttt{p\_sig\_cellular\_genes.csv}.}
\end{figure}

\section{Discussion}

\lipsum[9]

\section{Methods and Materials}

Guidelines can be included for standard research article sections, such as this one. 

\lipsum[3]

\section{Some \LaTeX{} Examples}
\label{sec:examples}

Use section and subsection commands to organize your document. 
\LaTeX{} handles all the formatting and numbering automatically. 
Use ref and label commands for cross-references.

\subsection{Figures and Tables}


If you use the following prefixes for your \verb|\label|:
%
\begin{description}
\item[Figures] \texttt{fig:}, e.g.~\verb|\label{fig:view}|
\item[Tables] \texttt{tab:}, e.g.~\verb|\label{tab:example}|
\item[Equations] \texttt{eq:}, e.g.~\verb|\label{eq:CLT}|
\item[Boxes] \texttt{box:}, e.g.~\verb|\label{box:simple}|
\end{description}
%
you can then use the convenience commands as in \verb|\FIG{cells}|, \ to generate cross-reference \FIG{view}. 

\subsection{Citations}

LaTeX formats citations and references automatically using the bibliography records in your .bib file, which you can edit via the project menu. 
Use the \verb|\cite| command for an inline citation, like \cite{trapnell2014pseudo}, and the \verb|\citep| command for a citation in parentheses \citep{trapnell2014pseudo}. 
The LaTeX template uses a slightly-modified Vancouver bibliography style. 
If your manuscript is accepted, the eLife production team will re-format the references into the final published form. 
\emph{It is not necessary to attempt to format the reference list yourself to mirror the final published form.}


\section{Acknowledgments}

Additional information can be given in the template, such as to not include funder information in the acknowledgments section.

\nocite{*} % This command displays all refs in the bib file
\bibliography{references}


\end{document}
