%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ELIFE ARTICLE TEMPLATE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% PREAMBLE 
\documentclass[9pt,lineno]{elife}
% Use the onehalfspacing option for 1.5 line spacing
% Use the doublespacing option for 2.0 line spacing
% Please note that these options may affect formatting.

\usepackage{lipsum} % Required to insert dummy text
\usepackage[version=4]{mhchem}
\usepackage{siunitx}
\DeclareSIUnit\Molar{M}
\usepackage{indentfirst} %I prefer to have indents following section headings, I find it to be more consistent in style throughout document
\usepackage{natbib}

\renewcommand{\floatpagefraction}{1}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE SETUP
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Transcriptional dynamics of influenza virus infection in single cells}

\author[1]{Alistair B. Russell}
\author[2]{Cole Trapnell}
\author[1,2*]{Jesse D. Bloom}
\affil[1]{Basic Sciences Division and Computational Biology Program, Fred Hutchinson Cancer Research Center, Seattle, United States}
\affil[2]{Department of Genome Sciences, University of Washington, Seattle, United States}
\corr{jbloom@fredhutch.org}{}

% \presentadd[\authfn{5}]{eLife Sciences editorial Office, eLife Sciences, Cambridge, United Kingdom}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE START
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\maketitle

\begin{abstract}
Influenza virus infection induces large changes in cellular transcription.
Previously this has mostly been looked at using bulk measurements
Here we examine the process at the level of single cells.
We find extremely wide variation in the extent of viral gene transcription across infected cells.
IFN induction is very rare.
Some cellular pathways may be consistently altered in cells with high burden of viral transcripts.
Overall, highlights remarkable heterogeneity in the outcome of infection.
\end{abstract}


\section{Introduction}

Heterogeneity is important in a lot of cellular processes even when isogenic~\citep{Shalek:2014ey,Shalek:2013ej}.

Population (genetic) heterogeneity is also important. 
Viral quasispecies, cancer single-cell, etc.
Salmonella paper (PhoP).

Literature on viral burst-size heterogeneity.
This goes back to Delbruck, Andino polio paper~\citep{Schulte:2014br}, the MDCK / flu paper.

Discuss segmented nature of influenza.
Maybe in the context of how this could further increase heterogeneity because there is a lot of potential for entire genes to be missing.
Includes Yewdell and Lowen papers.

\section{Results}

\subsection{Strategy to measure mRNA in single virus-infected cells.}
We performed single-cell mRNA sequencing using a droplet-based system that physically isolates individual cells prior to reverse transcription~\citep{Macosko:2015ht,Klein:2015ki}.
Each droplet is associated with a unique \emph{cell barcode} that tags all mRNAs from that cell during reverse-transcription.
A random \emph{unique molecular identifier (UMI)} is additionally appended to each mRNA molecule during reverse transcription.
The 3' ends of the mRNAs are sequenced and mapped to the human and influenza virus transcriptomes to determine transcript identities.
This information is combined with that provided by the UMIs and cell barcodes to quantify the number of molecules of each mRNA species that have been captured for each cell.

Infected cells will express viral as well as cellular mRNAs -- however the cell barcodes and UMIs cannot distinguish whether a cell was initially infected by one or multiple viral particles.
We therefore engineered an influenza virus (strain A/WSN/1933) that additionally carried \emph{viral barcodes} consisting of synonymous mutations near the 3' end of each transcript (Figure~\ref{fig:workflow}A).
Critically, these synonymous mutations did not greatly impact viral growth kinetics (Figure~\ref{fig:workflow}B).
We infected A549 human lung carcinoma cells with an equal mix of the wild-type and synonymously barcoded viruses.
Cells infected by a single virion will exclusively express mRNAs from either wild-type or synonymously barcoded virus, whereas cells that are co-infected with multiple virions will often express mRNAs from both the wild-type and synonymously barcoded viruses (Figure~\ref{fig:workflow}C).
\begin{figure}
\centerline{\includegraphics[width=0.8\linewidth]{figures/Workflow/workflow.pdf}}
\caption{\label{fig:workflow} Experimental design.
{\bf (A)}  
We engineered a virus that carried two synonymous mutations near the 3' end of each mRNA.
At top are the specific mutations for PB2.
At bottom are locations of the synonymous mutations relative to the typical distribution of read depth for our 3'-end sequencing.
{\bf (B)} 
The wild-type and synonymously barcoded viruses transcribe their genes with similar kinetics. 
Shown is the abundance of the viral hemagglutinin (HA) transcript relative to the cellular housekeeping gene L32 as assessed by qPCR in A549 cells infected at an MOI of 0.5 (as determined on MDCK-SIAT1 cells).
Error bars $\pm$ S.D., n=3.
{\bf (C)}  
A549 cells were infected with an equal mixture of mutant and wild-type virus. 
Immediately prior to collection, cells were physically separated into droplets and cDNA libraries were generated containing the indicated barcodes. 
The libraries were deep sequenced, and the data processed to create a matrix that gives the number of molecules of each transcript observed in each cell.
Infected cells were further annotated by whether their viral mRNAs derived from wild-type virus, synonymously barcoded virus, or both.
}
\figdata{Sequences of wild-type and barcoded viruses are in \texttt{viralsequences.fasta}.}
\end{figure}

We took care to generate stocks of virus that were relatively ``pure'' of defective particles.
Stocks of viruses typically contain an array of biologically active viral particles, some of which are defective for replication owing to mutations or deletions in essential viral genes ~\citep{Brooke:2014ch,Fonville:2015cg,Lauring:2010if,Dimmock:2014dk}.
These defective particles become prevalent when a virus is grown at high multiplicity of infection (MOI), where complementation permits the growth of otherwise deleterious genotypes ~\citep{Lopez:2014en}.
To minimize the levels of defective particles, we propagated our viral stocks at low MOI for a relatively brief period of time~\citep{Xue:2016bw}.
We then validated that our stocks exhibited greater purity of infectious particles than a stock propagated at high MOI by verifying that they had a higher ratio of infectious particles to both virion RNA (Figure~\ref{fig:viruspopulations}A) and particles capable of inducing expression of a single viral protein (Figure~\ref{fig:viruspopulations}B).
In addition, viral stocks with many defective particles are more immunostimulatory~\citep{Tapia:2013kf}.
We confirmed that our viral stocks induced much less interferon than a stock propagated at higher MOI (Figure~\ref{fig:viruspopulations}C).

\begin{figure}
\centerline{\includegraphics[width=0.7\linewidth]{figures/Validating_barcode_virus/validating_populations_D02.pdf}}
\caption{\label{fig:viruspopulations} The viral stocks in our experiments are relatively pure of defective particles. 
{\bf (A)}
Our viral stocks have a higher ratio of infectious particles to HA virion RNA compared to a high-defective stock propagated at high MOI.
HA viral RNA was quantified by qPCR on virions. 
Error bars $\pm$ S.D., n=6 (qPCR replicates). 
{\bf (B)} 
Our viral stocks have a higher ratio of infectious particles to particles capable of expressing the viral HA protein.
A549 cells were infected with all three viral stocks at an MOI of 0.1, and the percentage of cells expressing HA protein at 9 hours post-infection was quantified by antibody staining and flow cytometry.
{\bf (C)} 
Our viral stocks are less immunostimulatory than virus propagated at high MOI. 
Measurements of \textit{IFNB1} transcript by qPCR normalized to the housekeeping gene L32 in A549 cells at 10 hours post infection at an MOI of 0.5.
MOIs were calculated by TCID50 on MDCK-SIAT1 cells.
Error bars $\pm$ S.D., n=3}
\figsupp[Full flow cytometry data for panel B . \label{figsupp:HAstain}]{
Full flow cytometry date for  Figure~\ref{fig:viruspopulations}B. All populations infected at an MOI of 0.1 as calculated by TCID50 on MDCK-SIAT1 cells. 
{\bf (A)} Uninfected gating control
{\bf (B)} Cells infected with wild-type virus
{\bf (C)} Cells infected with synonymously barcoded virus
{\bf (D)} Cells infected with a stock of wild-type virus propagated at a high MOI, therefore enriched in defective virus.
}{\includegraphics[width=0.7\linewidth]{figures/Validating_barcode_virus/HA_stain_supplement.png}}\end{figure}

\subsection{Single cells show a wide range of expression of viral transcripts.}
We infected A549 cells at low MOI with a mixture of the wild-type and synonymously barcoded viruses, and collected cells for sequencing at 6, 8, and 10 hours post-infection, including a replicate of the 8-hour timepoint.
We recovered between 3,000 and 4,000 cells for each sample (Figure~\ref{fig:cells}A). 
As expected for a low-MOI infection, most cells expressed little or no viral mRNA (Figure~\ref{fig:cells}B).
Also as expected, the amount of viral mRNA per cell among infected cells increased over time (Figure~\ref{fig:cells}B).
But what was most notable was how widely the number of viral mRNA molecules varied among infected cells.
While the fraction of mRNA derived from virus was $<$0.1\% for most cells, viral mRNA constituted close to half the transcriptome in a few cells at 8 and 10 hours (Figure~\ref{fig:cells}C).
%Therefore, there is extraordinary variability in the number of viral transcripts per infected cell.

\begin{figure}
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_cell_mRNA_summary.pdf}}
\caption{\label{fig:cells}
There is a very wide distribution in the number of viral mRNAs detected per cell.
{\bf (A)} 
Number of cells sequenced for each sample.
{\bf (B)} 
Cumulative distribution of the fraction of mRNA derived from influenza virus for each sample.
For each point on the x-axis, the y-axis indicates the fraction of cells with at least that much of their mRNA from virus.
{\bf (C)} 
The number of cellular and viral mRNAs for each cell is plotted as a point.
The blue lines show the overall distribution of the number of cellular mRNAs per cell.
Cells outside the red lines were considered outliers (possibly derived from two cells per droplet), and were excluded from subsequent analyses.
}
\end{figure}

A complicating factor is that uninfected cells could have small amounts of viral mRNA due to leakage of transcripts from lysed cells.
It is therefore important to establish a threshold for identifying truly infected cells.
We can do this by taking advantage of the fact that roughly half the infecting virions bear synonymous barcodes.
Reads derived from lysed cells will be drawn from both wild-type and synonymously barcoded viral transcripts.
However, most cells are infected by at most one virion, and so the reads from truly infected cells will usually derive almost entirely from one of the two viral variants. 
Figure~\ref{fig:viralbarcodes}A shows the fraction of viral reads in individual cells from each viral variant, and Figure~\ref{fig:viralbarcodes}B indicates the fraction of viral reads from the most abundant variant in that cell.
Most cells with large amounts of viral mRNA have viral transcripts exclusively derived from one viral variant -- indicating non-random partitioning as expected from viral infection.
However, cells with a small amount of viral mRNA often have viral transcripts from both variants, as expected from the random partitioning associated with simple mRNA leakage.
Finally, a few cells with large amounts of viral mRNA have viral transcripts from both variants, reflecting co-infection with both variants.

We determined the threshold amount of viral mRNA per cell at which the barcode partitioning clearly resulted from infection rather than leakage (Figure~\ref{fig:viralbarcodes}C), and used this threshold to annotate cells that we were confident were truly infected.
We also annotated as co-infected cells above this threshold that had mRNA from both viral variants.
Figure~\ref{fig:viralbarcodes}D shows the number of cells annotated as infected and co-infected for each sample -- these cells are just a small fraction of the number of cells with any viral read.
These annotation thresholds are conservative, and will miss some true low-level infections, as well as any co-infections with the same viral variant.
However, it is important that the analyses below are restricted to cells that are truly infected with virus, so we accepted the loss of some low-level infections in order to avoid false positives.
Because the bulk of cells are not infected, we subsampled the uninfected cells to the numbers shown in Figure~\ref{fig:viralbarcodes}D to balance the proportions of infected and uninfected cells for all subsequent analyses.

Strikingly, the extreme variation in the number of viral transcripts per cell remains even after we apply these rigorous criteria for annotating infected cells (Figure~\ref{fig:viralbarcodes}E). 
The fraction of viral mRNA per infected cell follows a roughly exponential distribution, with many cells having few viral transcripts and a few cells having many.
In fact, a bare few infected cells contribute the majority of influenza transcripts.
Specifically, at 6 and 8 hours $<$10\% of infected cells are responsible for over half the viral transcripts, while at 10 hours $\approx$15\% of infected cells produce over half the viral transcripts (Figure~\ref{fig:viralbarcodes}-Figure~supplement~\ref{figsupp:cumulfracflu}).

\begin{figure}[t!]
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_frac_flu_summary.pdf}}
\caption{\label{fig:viralbarcodes}
Synonymous barcodes on the viral mRNAs distinguish true infections from cells that contain viral mRNAs derived from leakage of lysed cells.
{\bf (A)}
Cells with at least two viral mRNAs for which the barcode could be called, arranged in order of increasing influenza transcript counts.
Bar heights denote the number viral mRNAs on a log\textsubscript{10} scale, bar coloring is linearly proportional to the fractions of viral mRNAs derived from wild-type and synonymously barcoded virus.
{\bf (B)}
Same as (A), but each bar is colored according to the relative fraction of the more common (major) and less common (minor) virus variant.
At low levels of viral mRNA there is often a roughly equal mix, suggesting contamination with viral mRNAs leaked from lysed cells.
At higher levels of viral mRNA, cells generally have only one viral variant, suggesting infection initiated by a single particle.
A few cells are also obviously co-infected with both viral variants.
{\bf (C)}
We determined a cutoff for calling ``true'' infections by fitting a curve to the mean barcode purity of all cells with greater than a given fraction of their mRNA derived from virus.
We called the cutoff at the point at which purity stops increasing with the fraction of viral mRNA.
{\bf (D)}
The number of cells identified as infected and co-infected for each sample, as well as the number of cells with any viral read.
For all subsequent analyses, we subsampled the number of uninfected cells per sample to the greater of 50 or the number of infected cells.
{\bf (E)} 
Distribution of the fraction of mRNA per cell derived from virus for both infected and co-infected cells.
}
\figsupp[Number of viral barcodes called.\label{figsupp:barcodescalled}]{
The number of viral barcodes called for each sample and influenza gene segment. 
Viral transcripts are classified as \emph{syn} if they mapped to a synonymously barcoded influenza transcript, \emph{wt} if they mapped to a wild-type influenza transcript, \emph{invalid} if multiple reads for the same UMI differed on the status of the viral barcode, and as \emph{uncalled} if none of the reads for that UMI overlapped the region of the viral transcript containing the viral barcode.
}{\includegraphics[width=\linewidth]{figures/synbarcodes_umistats.jpg}}
\figsupp[Fraction of total viral mRNA derived from a given fraction of infected cells.\label{figsupp:cumulfracflu}]{
The total fraction of all viral mRNA among infected cells that is attributable to a given fraction of these cells.
For instance, the plot for the 8-hour sample shows that roughly 50\% of all viral mRNA is derived from 10\% of the infected cells.
}{\includegraphics[width=\linewidth]{figures/p_cumul_flu.pdf}}
\end{figure}

\subsection{Absence of viral genes partially explains cell-to-cell variability in viral transcript abundance.}
The influenza genome is segmented, and cells can fail to express a viral mRNA if the encoding gene segment is not packaged in the infecting virion or fails to initiate transcription after infection.
Indeed, several groups have reported that the majority of infected cells fail to express at least one viral gene~\citep{Brooke:2013kb, Heldt:2015iz}. 
We wondered if the absence of specific viral genes might be associated with reduced amounts of viral mRNA within single infected cells.
In particular, transcription of influenza virus mRNAs is performed by the viral ribonucleoprotein (RNP) complex, which consists of the three proteins that encode the tripartite polymerase (PB2, PB1, and PA) as well as nucleoprotein (NP).
Each viral gene segment is associated with one RNP in incoming infecting virions, but secondary transcription by newly synthesized RNPs requires the presence of the viral genes for each of the four constituent RNP proteins so that the cell can produce more RNP proteins~\citep{Vreede:2004ip}.
This secondary transcription is a major source of viral mRNAs, as evidenced by the fact that blocking synthesis of the RNP proteins reduces the amount of viral mRNA by several orders of magnitude in bulk cells (Figure~\ref{fig:fluburdenbyflugene}-Figure~supplement~\ref{figsupp:cyclohexamide}).

\begin{figure}[t!]
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_flu_burden_flu_gene_merge.pdf}}
\caption{\label{fig:fluburdenbyflugene}
The absence of viral genes explains some of the variability in the amount of viral mRNA per cell.
{\bf (A)} 
Fraction of mRNA in each infected cell derived from virus as a function of the normalized expression of each viral gene in that cell, taken over all time points.
Cells with high viral burden always express all the RNP genes, but some cells with high viral burden lack each of the other viral genes.
{\bf (B)}
Box plots showing the per-cell viral burden among cells with $>$0.5\% of their mRNA from virus, binned by whether or not the cells express each viral gene.
A Wilcoxon signed-rank test was used to test the null hypothesis that absence of each gene does not affect viral burden: **** = $P < 10^{-4}$, *** = $P < 10^{-3}$,  * = $P < 0.05$.
Similar results are obtained if we examine only the 10-hour timepoint (Figure~\ref{fig:fluburdenbyflugene}-Figure~supplement~\ref{figsupp:10hrfluburdenbyflugene}).
Top and bottom of box bounds first and third quartiles of dataset, line dividing the box is at the median observation.
Whiskers extend to the highest or lowest point within 1.5x the interquartile range, outliers shown as circles.
{\bf (C)}
Fraction of infected cells that express all four RNP genes and also express each of the four other genes, as well as the fraction that express \emph{all} four of the other genes.
The fraction of cells that express all four other genes is well predicted by simply multiplying the frequencies of cells that express each of these genes individually, indicating that gene absence is approximately independent across these four genes.
}
\figsupp[Secondary transcription from newly synthesized RNPs is a major source of viral mRNA during bulk infections.\label{figsupp:cyclohexamide}]{
A549 cells were infected at an MOI of 0.2 as calculated on MDCK-SIAT1 cels in either the presence or absence of the protein-translation inhibitor cyclohexamide, and viral mRNA was quantified at 8 hours post-infection by qPCR.
The cyclohexamide prevents translation of new PB2, PB1, PA, and NP protein, and so prevents the formation of the new RNPs needed for secondary transcription.
The bars show the relative amount of HA and PB2 mRNA in the absence versus the presence of cyclohexamide. Error $\pm$ S.D. n=3. 
}{\includegraphics[width=0.3\linewidth]{figures/Primary_transcription_measurements/Cyclohexamide.pdf}}
\figsupp[Like panel (B) but for the 10-hr sample only.\label{figsupp:10hrfluburdenbyflugene}]{
The absence of viral RNP genes but \emph{not} non-RNP genes remains significantly associated with reduced viral burden when we examine only the 10-hr sample, which is the single time point with the most data points.
The difference for NP is no longer statistically significant due to low counts of infected cells lacking NP, but the trend remains.
}{\includegraphics[width=\linewidth]{figures/p_10hr_flu_burden_flu_gene_test}}
\figdata{The numerical data for panel (C) are in \texttt{p\_missing\_genes.csv}.}
\end{figure}

We examined the total fraction of mRNA in each cell that derives from virus versus the relative expression of each viral gene (Figure~\ref{fig:fluburdenbyflugene}A). 
Cells that lack an RNP gene never derive more than a few percent of their mRNAs from virus, demonstrating that the presence of these RNP genes is essential for high levels of viral transcription.
However, we observe cells that lack each of the other non-RNP genes but still derive $\approx$40\% of their mRNAs from virus, suggesting that none of the other genes are important for high levels of viral transcription.
These results are statistically supported by Figure~\ref{fig:fluburdenbyflugene}B, which shows that absence of any RNP gene but \emph{not} any of the other viral genes is associated with reduced amounts of viral mRNA. 
However, gene absence clearly does not explain all of the variability in viral gene expression, since even cells expressing all viral genes exhibit a very wide distribution in the amount of viral mRNA that they express. 
The rest of this variability must be due to additional mechanisms such as differences in host-cell state, viral mutations not detectable by our 3'-end sequencing approach, or inherent stochasticity.  

We also sought to quantify the fraction of infected cells that completely failed to express a given gene.
We limited this analysis to examining the presence / absence of the non-RNP genes in cells expressing all four RNP genes, since we might fail to detect viral transcripts that are actually present at low levels in RNP-deficient cells due to the lower viral burden in these cells.
At the 8- and 10-hour time points, between 5\% and 17\% of cells fail to express any one of the four non-RNP genes (Figure~\ref{fig:fluburdenbyflugene}C).
The absence of a given gene appears to be an independent event, as the probability of observing all four non-RNP genes in a cell is well predicted by simply multiplying the probabilities of observing each gene individually (Figure~\ref{fig:fluburdenbyflugene}C). 
Our measurements of the frequency at which infected cells fail to express individual viral genes are roughly consistent with estimates made by others using different means~\citep{Brooke:2013kb,Dou:2017cp}.
	
\subsection{The relative amounts of different viral mRNAs are fairly consistent across cells.}
The results above show that the total amount of viral mRNA in infected cells varies over orders of magnitude.
Does the relative expression of viral genes exhibit similar cell-to-cell variability?
To address this question, we focused on cells that derived $>$5\% of their mRNA from virus, since estimates of relative viral gene expression will be less noisy in cells with more viral mRNAs.

\begin{figure}
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_flu_expr_aledit.pdf}}
\caption{\label{fig:fluexpr}
Relative expression of influenza virus genes in highly infected cells (>5\% of total mRNA from virus).
{\bf (A)} 
The fraction of viral mRNA from each viral gene for each cell. 
{\bf (B)}
Box plots showing the distribution of the fraction of viral mRNA per cell from each viral gene.
The black lines at the notches are the medians, and the tops and bottoms of boxes indicate the first and third quartiles.
Whiskers extend to the highest or lowest data point observed within 1.5x the interquartile range, outliers shown as circles.
Notches extend 1.58x the interquartile range divided by the square root of the number of observations. 
{\bf (C)}, {\bf (D)} 
The same plots, but only including cells for which we observed at least one molecule of each viral gene.
}
\figdata{The raw data are in \texttt{p\_flu\_expr.csv}.}
\end{figure}

In contrast to the extreme variability in the total viral mRNA per cell, the fraction of this mRNA derived from each gene is much more consistent across cells (Figure~\ref{fig:fluexpr}A).
Total viral mRNA varies by orders of magnitude, but the fraction from any given viral gene is fairly tightly clustered around the median value for all cells (Figure~\ref{fig:fluexpr}B).
The relative levels of each viral mRNA in our cells are similar to prior bulk measurements made by Northern blots~\citep{Hatada:1989vz}, which also found an expression hierarchy of M $>$ NS $\gg$ NP $>$ NA $>$ HA $\gg$ PB2 $\sim$ PB1 $\sim$ PA.
The cell-to-cell consistency in the relative expression of different viral genes generally becomes even tighter if we limit the analysis only to cells that express all eight viral genes (Figure~\ref{fig:fluexpr}C,D).
Therefore, with the exception of complete gene absence, the factors that drive the dramatic cell-to-cell variability in the amount of viral mRNA appear to have roughly similar effects on all viral genes in a given cell.
This finding is consistent with prior work showing positive correlations among the abundance of several viral gene segments in individual cells.

\subsection{Co-infection can provide infected cells with the full complement of viral genes.}
Our sequencing enables us to identify the rare cells that were co-infected with both wild-type and synonymously barcoded viral variants.
Overall, we captured 10 such co-infected cells that had $>$5\% of their mRNA derived from virus (Figure~\ref{fig:coexpression}).
Seven of these 10 cells expressed all eight viral genes.
Remarkably, the majority (4 of 7) of these cells would \emph{not} have expressed all the viral genes in the absence of co-infection, since they have at least one gene exclusively derived from each viral variant.
For instance, the cell with 11.2\% of its mRNA from virus in the upper right of Figure~\ref{fig:coexpression} expresses M only from the wildtype viral variant, and NP and HA only from the synonymously barcoded variant.
Our data therefore provide direct single-cell support for the idea that co-infection can rescue missing viral genes~\citep{Brooke:2013kb,Fonville:2015cg}.

\begin{figure}
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_coinfection.pdf}}
\caption{\label{fig:coexpression}
The abundance of each viral transcript in cells that are co-infected with the two viral variants and have $>$5\% of their mRNA derived from virus.
The bars show the logarithm of the numbers of each viral mRNA detected, and are colored in linear proportion to the fraction of that mRNAs derived from wild-type or synonymously barcoded virus.
}
\figsupp[Co-infected cells express roughly equal amounts of a gene from each infecting viral variant. \label{figsupp:flowcyto}]{
{\bf(A)} Cells were co-infected with a mix of wild-type virus and virus in which the HA gene was replaced by GFP flanked by the terminal regions of the HA gene segment. 
At 10 hours post-infection, cells were analyzed by flow cytometry for HA and eGFP expression.
{\bf(B)} The expression of HA and GFP are correlated in co-infected cells. 
Shown are the quantile-normalized HA and eGFP signals for double-positive cells. 
Cells are colored by density, using a Gaussian kernel density estimate. 
{\bf(C)},{\bf(D)},{\bf(E)} Gating controls, single infection with eGFP virus, single infection with wild-type virus, and uninfected cells, respectively. 
}{\includegraphics[width=\linewidth]{figures/Pseudovirus_flow_cytometry/coinfFlow_D02.png}}
\figdata{The raw data plotted in this figure are in \texttt{p\_co-infection.csv}.}
\end{figure}

Another observation from Figure~\ref{fig:coexpression} is that co-infected cells usually express roughly equal amounts of transcripts from each of the two viral variants.
This observation is consistent with the finding by \citet{Dou:2017cp} and \citet{Huang:2008gy} that the temporal window for co-infection is short -- if both viral variants infect a cell at about the same time, then neither will have a headstart and so each will have a roughly equal opportunity to transcribe its genes.

To support this idea with a larger dataset albeit at lower resolution, we generated a virus in which the HA coding sequence was replaced by GFP.
We then co-infected cells with a mix of wildtype and $\Delta$HA-GFP virus and used flow cytometry to score cells for the presence of HA only (infection by wildtype virus), GFP only (infection by $\Delta$HA-GFP virus), or both (co-infection) as shown in Figure~\ref{fig:coexpression}~Figure~supplement~\ref{figsupp:flowcyto}.
As in our single-cell sequencing data, we found that expression of HA and GFP were highly correlated, indicating that co-infected cells typically expressed roughly equal amounts of transcript from each viral variant.

\subsection{Activation of the anti-viral response is rare in single infected cells.}
Because our sequencing captured all polyadenylated transcripts, we can examine whether there are prominent changes in the host-cell transcriptome in sub-populations of infected cells.
In particular, influenza virus infection can trigger innate-immune sensors that lead to the transcriptional induction of type I and III interferons, and subsequently of anti-viral interferon-stimulated genes~\citep{Killip:2015dw,Iwasaki:2014dw,Crotta:2013ef}.
In bulk cells, this interferon response is one of the most prominent transcriptional signatures of influenza-virus infection~\citep{Geiss:2002a}.
However, activation of the interferon response is stochastic and bi-modal at the level of single cells~\citep{Chen:2010cr,Shalek:2014ey,Shalek:2013ej}.
We therefore hypothesized that we might see two sub-populations of infected cells: one in which the interferon response inhibited viral transcription, and another in which the virus was able to express high levels of its mRNA by evading or blocking this response.

\begin{figure}
\centerline{\includegraphics[width=0.8\linewidth]{figures/p_small_tsne_merge.pdf}}
\caption{\label{fig:tsne}
A t-SNE plot created by semi-supervised clustering of the cells using genes that co-vary with viral infection status.
Each point is a single cell, and each panel shows an identical layout but colors the cells according to a different property.
{\bf (A), (B)}
Cells colored by the fraction of their mRNA that is derived from virus.
{\bf (C)}
Cells colored by the experimental sample from which they are derived.
{\bf (D), (E)}
Cells colored by the number of detected transcripts from type I and III interferons (IFN).
Only one cell has detectable interferon expression.
{\bf (E)}
Cells colored by the expression of the interferon-stimulated gene IFIT1.
{\bf (F)}
Cells colored by whether they express the viral NS gene.
The one interferon-positive cell is lacking NS, but many interferon-negative cells also lack NS.
}

\end{figure}

To examine whether there were distinct sub-populations of virus-infected cells, we used a semi-supervised t-SNE approach to cluster cells by genes that co-varied with viral infection status \citep{Trapnell:2014kg,VanderMaaten:2008tm}.
As shown in Figure~\ref{fig:tsne}A,B, this approach effectively grouped cells by the amount of viral mRNA that they expressed.
Sample-to-sample variation was regressed away during the clustering, as cells did not obviously group by time-point, with expected exception that the uninfected and 6-hour samples had few cells in the region of the plot corresponding to large amounts of viral mRNA (Figure~\ref{fig:tsne}C).

But to our surprise, we did not see a prominent clustering of infected cells into sub-populations as expected if the interferon response was strongly activated in some cells.
To investigate further, we annotated each cell by the total number of type I and III interferon transcripts detected.
Remarkably, only a single cell expressed detectable interferon (Figure~\ref{fig:tsne}D).
We also examined interferon-stimulated genes, which are induced by autocrine and paracrine interferon signaling.
Figure~\ref{fig:tsne}E shows expression of one such gene, IFIT1~\citep{Fensterl:2011fp}.
As with interferon itself, expression of IFIT1 was rare and most prominent in the single interferon-positive cell, presumably due to the higher efficiency of autocrine versus paracrine signaling.
Notably, interferon and interferon-stimulated genes were also relatively ineffective at blocking viral transcription in the single cell in which they were potently induced, since $>$10\% of the mRNA in this cell was derived from virus (Figure~\ref{fig:tsne}A,B,D,E).

We posited that the paucity of interferon induction might be due to the activity of influenza virus's major interferon antagonist, the NS1 protein.
We therefore identified cells that expressed substantial amounts of viral mRNA but lacked the NS gene (Figure~\ref{fig:tsne}F).
Consistent with the idea that NS1 is important for suppressing interferon, the one interferon-positive cell lacked detectable expression of the NS gene.
But other cells that lacked NS expression still failed to induce a detectable interferon response, despite often having $>$10\% of their mRNA derived from virus (Figure~\ref{fig:tsne}).
Therefore, most infected cells fail to express interferon even when the virus lacks its major known interferon antagonist.
Our results are in line with other work showing that NS1-deficient influenza virus fails to deterministically induce interferon~\citep{Killip:2017ef,Kallfass:2013kp}.
Therefore, single cells infected at low MOI with a relatively ``pure'' stock of influenza virus clearly activate innate-immune responses much less frequently than might be presumed from bulk studies that flood cells with viral stocks full of defective particles.

\subsection{Genes involved in the oxidative-stress response co-vary with viral gene expression.}
We next examined whether any host genes were differentially expressed in cells with more viral mRNA.
We restricted this analysis to infected cells with all eight viral genes in order to focus on cellular genes that were associated with viral mRNA burden independent of any effects due to the stochastic presence or absence of particular viral transcripts.
We identified 45 cellular genes that co-varied with viral gene expression at a false discovery rate of 0.1 (Figure~\ref{fig:cellulargenes}).

\begin{figure}
\centerline{\includegraphics[width=0.7\linewidth]{figures/p_cellular_heatmap.pdf}}
\caption{\label{fig:cellulargenes}
Cellular genes that co-vary in expression along with the the amount of influenza virus mRNA in individual cells expressing all eight viral genes.
Shown are all genes that are differentially expressed at a false discovery rate of 0.1. 
Genes for which the color goes from blue at left to red at right are expressed at higher levels in cells with more viral mRNA. Scale is {\color{red} not sure what to put here} }
\figdata{The full results of the differential expression test is in \texttt{p\_sig\_cellular\_genes.csv}.}
\figdata{The results of a gene-set analysis are in \texttt{p\_sig\_cellular\_genes.csv}.}
\figsupp[Many genes that co-vary with the amount of viral mRNA are involved in the response to oxidative stress. \label{figsupp:ROStable}]{
Table delineating genes in (Figure~\ref{fig:cellulargenes}) that are associated with the response to oxidative stress ~\citep{Duong:2017ju,Jung:2017jz,Lee:2017ba,Peuchant:2017hj,MacLeod:2016kt,Jiang:2016cf,Gorrini:2013jv,Miura:2013fr,Kim:2009gi,Banning:2005cz,Murray:2003ie,Doyle:1999ts}.
}{\includegraphics[width=0.7\linewidth]{figures/ROStable.pdf}}
\end{figure}

Many of the genes that exhibit increased expression in cells with large amounts of viral mRNA are known or suspected to be regulated by the Nrf2 master regulator in response to oxidative stress.
These genes produce proteins that are involved in detoxification of reactive oxygen species or resultant products, the management of misfolded proteins, the electron transport chain, or a general stress response (Figure~\ref{fig:cellulargenes}~Figure~supplement~\ref{figsupp:ROStable}). 
We additionally see reduced expression of the nitric oxide synthase interacting protein (NOSIP). 
Transient oxidative stress is known to occur during viral infection, and may act in a proviral fashion via MAPK activation driving vRNP export~\citep{Amatore:2014cs}.
The antioxidant response is thought to be largely antiviral, potentially through inhibition of MAPK activity \citep{Lin:2016ec,Sgarbanti:2014ht}.
Our data do not reveal whether the expression of genes involved in the response to oxidative stress are a cause or a symptom of higher levels of viral mRNA, and further investigation of this topic is an interesting area for future work.
Overall, the expression of some of the host genes in Figure~\ref{fig:cellulargenes}, along with inherent stochasticity and specific mutations to viral genes in individual cells, probably explain most of the variation in viral mRNA levels observed across the infected cells in our dataset.

\section{Discussion}
Still needs to be written.

\section{Methods and Materials}

\subsection{Tissue culture and viral rescue}
Cell lines used in this study were: the human lung epithelial carcinoma line, A549, obtained from ATCC, the canine kidney cell line overexpressing human SIAT1, MDCK-SIAT1, obtained from Sigma Aldrich, and the human embryonic kidney cell line, 293t, obtained from ATCC. 
All cells were maintained in D10 (DMEM supplemented with 10\% heat-inactivated FBS, 2 mM L-glutamine, 100 U of penicillin/ml, and 100 \si{\micro}g of streptomycin/ml) at 37 \si{\degreeCelsius} at 5 \% CO\textsubscript{2}.
We generated viral stocks using a previously-defined reverse-genetics approach~\citep{Neumann:1999ws}. 
In brief, to generate wild-type virus we transfected bidirectional plasmids encoding the eight segments of A/WSN/1933 (kindly provided by Robert Webster of St. Jude Children's Research Hospital) into a coculture of 293t and MDCK-SIAT1 cells seeded at a ratio of 8:1. 
At 24 hours post-transfection, we changed media from D10 to influenza growth media (IGM, Opti-MEM supplemented with 0.01\% heat-inactivated FBS, 0.3\% BSA, 100 U of penicillin/ml, 100  \si{\micro}g of streptomycin/ml, and 100 \si{\micro}g of calcium chloride/ml).
At 48 hours post-transfection we harvested the virus-containing supernatant, removed cellular material by centrifugation at 300 x g's for 4 minutes, and stored aliquots at -80 \si{\degreeCelsius }. 
We then determined the TCID50 of these stocks on MDCK-SIAT1 cells.
Viral aliquots were then used to initiate an infection at an MOI of 0.01 on MDCK-SIAT1 cells in order to expand viral stocks and provide partial purging of defective viruses.
Viral supernatant was harvested at 36 hours post-infection, prior to the onset of noticeable cytopathic effect. 
Aliquots were again frozen at -80  \si{\degreeCelsius }, titered, and used in our single-cell experiments.
Synonymously barcoded virus was generated identically to wild-type viral stocks, using plasmids encoding the synonymously-encoded viral segments described in (supplemental data).
Mutant virus containing a gene encoding eGFP in place of the HA gene segment was generated using previously-described methods for growing HA-deficient virus, utilizing MDCK-SIAT1 cells expressing HA from a constituitive CMV promoter~\citep{Doud:2016gm}.
A unidirectional vector for generating viral vRNA encoding eGFP flanked by HA packaging sequences was generated from a previously described vector, pHH21~\citep{Neumann:1999ws}. 
In order to obtain sufficient titers, HA-eGFP virus was expanded for 44 hours after initiating infection at an MOI of 0.01 rather than 36 hours.

\subsection{qPCR measurements of infection kinetics}
For qPCR measurements of viral transcriptional kinetics and interferon induction, A549 cells were seeded at 3 x 10\textsuperscript{5} cells / well in a 6-well tissue culture plate in D10 the day prior to infection. 
On the day of infection, a single well was trypsonized and enumerated, providing a measurement for MOI calculation.
Immediately before infecting, D10 was replaced with IGM and the indicated treatment applied.
For cells incubated with cycloheximide, the compound was added to a final concentration of 50  \si{\micro}g/ml at the time of infection -- previously confirmed to be sufficient to halt viral protein production~\citep{Killip:2014kz}.
RNA was purified using the QIAGEN RNeasy plus mini kit following manufacturer's suggestions, and purity and concentration was measured using UV absorbance prior to cDNA synthesis. 
cDNA was synthesized using an oligoDT primer and the SuperScript\textsuperscript{TM} III first-strand synthesis supermix from ThermoFisher using manufacturer's directions. 
Transcript abundance was measured using SYBR\textsuperscript{TM}  green PCR master mix, using a combined anneal/extension step of 60 \si{\degreeCelsius } for one minute with the primers listed in (Supplemental table).
Biological triplicates were performed for all samples.
For measurements of viral genomic HA content, vRNA was harvested from 80 \si{\micro}l of viral supernatant by the addition of 600 \si{\micro}l of RLT plus before proceeding with the standard QIAGEN Rneasy plus miniprep kit protocol, and cDNA was generated using  SuperScript\textsuperscript{TM} III first-strand synthesis supermix with primers specific for vRNA using the manufacturer's protocol for random-hexamer priming~\citep{Xue:2017dl,Hoffmann:2001vj}.
qPCR was then performed as for mRNA measurements.
In order to obtain molecular resolution on our measurements of HA vRNA, a standard curve was generated from three independent dilutions of the HA-encoding bidirectional plasmid used in our viral rescue. 
All vRNA values represent three independent RNA extractions with two replicate qPCR measurements. 

\subsection{Flow cytometric analyses of infection}
A549 cells were seeded in a 6-well plate and infected as described for qPCR analyses.
Cells were harvested by trypsonization, resuspended in phosphate-buffered saline supplemented with 2\% heat-inactivated FBS, and stained with 10  \si{\micro}g/ml of H17-L19, a mouse monoclonal antibody confirmed to bind to WSN HA in a prior study~\citep{Doud:2017bw}.
After subsequent washes in supplemented PBS, cells were stained with a goat anti-mouse IgG antibody conjugated to APC.
Cells were then washed, fixed in 1\% formaldehyde, and washed further before a final resuspension and analysis. 
After gating for single-cells in FlowJo, data were subsequently extracted using the R package, flowCore~\citep{LeMeur:2007uo}.
Extracted data were then analyzed using a custom Python script, and gates determined.
Guassian kernel density estimates were obtained using the scipy stats package method, guassian\_kde, using automatic bandwidth determination~\citep{vanderWalt:2017dp}.

\subsection{Generation of single cell infection data}
All samples were processed using the 10x Chromium Single Cell 3' platform, both in initial, tagged, cDNA generation and in final library preparation.
All time points, save for the 8 hour repeat, were performed on the same day as described below.
A549 cells were seeded as described above in a 6-well plate, with two wells per time point. 
A single well of cells was trypsonized and counted prior to initiation of the experiment for the purposes of calculating MOI.
Wild-type and synonymously barcoded virus were mixed to an estimated ratio of 1:1 based on prior, exploratory, single-cell analyses (data not shown). 
At the initiation of our experiment, all cells were changed from D10 to IGM, regardless of the desired time point.
Cells were then infected with an intended, flow cytometry calculated, MOI of 0.3, in order of time point -- 10 hour, then 8 hour, then 6 hour time points, spaced two hours apart.
This MOI was calculated in the absence of an additional washing step, and so potentially represents an overestimate of our final infectious dose.
One hour post infection, media was changed to fresh IGM.
All cells were then harvested for single-cell analysis concurrently -- ensuring all had spent equivalent time in changed media and normalizing potential batch effects across samples.
For our replicate 8 hour time point, cells were infected as above, save for no wash step was included, the flow cytometry informed MOI was instead 0.1, and the sample was prepared on a different day.
After harvest, cells were counted using disposable hemocytometers and diluted to equivalent concentrations with an intended capture of 3000 cells/sample following the guidelines provided by 10x genomics.
All subsequent steps followed 10x genomics guidelines through library preparation.
Samples were sequenced on an Illumina HiSeq. 

\subsection{Computational analysis of single cell transcriptomes}
Sequencing data were aligned and a gene-cell-barcode matrix determined using the 10x Genomics software package, CellRanger v. 2.0.0. 
Wild-type and synonymous viral transcripts were identified using a custom python script, and these annotations appended to the gene-cell-barcode matrix for further analysis.
These matrices were then evaluated using the R package Monocle, version 2.4.0~\citep{Qiu:2017dw,Trapnell:2014kg}.
Cell barcodes associated with 2.5-fold fewer or more UMI counts than the sample median were considered to be potential doublets or inefficient/empty reactions, and were excluded from further downstream analyses.
In order to empirically determine a cutoff for potential lysis-derived influenza reads we used the 10 hour time point, which had the highest influenza transcript count.
The relative purity of barcodes was calculated out to the 35th percentile of influenza-associated reads.
These values were plotted against the number of UMIs attributed to influenza, and a curve was fitted using the loess function in R. 
The derivative of the resultant curve was calculated, and the threshold set at the minimum value wherein the derivative is less than 0 -- indicating the point below which there is a sacrifice in sample purity, indicating an increase in the relative proportion of extracellular-derived influenza reads.
Cells meeting this threshold were defined as infected, and those failing to do so were excluded from further analysis.
For each time point an additional number of cells, either 50 or the number of infected cells determined by our threshold above, whichever was larger, with no detectable influenza transcripts were also included for further analysis.

Using Monocle'?s cell hierarchy function, cells were binned into those with no influenza, and those with reads derived from influenza. 
Influenza infected cells were further described to possess <2\%, >2\%, and >20\% of reads derived from influenza using nested cell classes. 
Candidate marker genes for t-distributed stochastic neighbor embedding (tSNE) dimensionality reduction that would allow us to separate these populations were then determined using the Monocle function markerDiffTable, excluding the effects of sample variation and the number of genes identified in a given cell, using a q-value cutoff of 0.01.
The specificity of these markers was determined using the function calculateMarkerSpecificity -- the top 40 markers were retained, and used to place populations in a two-dimensional plane based on tSNE dimensionality reduction.

Analyses of host expression required alteration to the standard Monocle pipeline and subsetting of our dataset.
First, we only considered host genes in our analyses, as the massive contribution of viral variation could swamp out more nuanced effects.
Second, as influenza has the capacity to suppress the host transcriptome ~\citep{BercovichKinori:2016iw}, and moreover contributes significantly to the total number UMIs that are attributed to a cell, we decided to assign the cell's size factor, a scalar value representing efficiency of UMI capture, based on host transcripts alone. 
We excluded uninfected cells from these analyses, as well as those influenza-infected cells that lacked a gene from any of the eight segments. 
Lastly, cells which contained a fraction of influenza-derived reads of greater than 25\% were assessed at a value of 25\% such that they did not contribute an undue weight to our analysis. 
Genes whose expression co-varied with the fraction of mRNAs in a cell derived from influenza were then identified, using the Monocle function differentialGeneTest, after removing variance explained by sample to sample variation. 
Retaining those meeting a q-value threshold of 0.1, the expression of these genes was visualized on a per-cell basis, arranging cells in order of increasing percentage of reads derived from influenza.

As an additional analysis, we ranked cells within each sample according to the relative percentage of reads derived from influenza.
As above, we only considered cells expressing genes from all eight viral genomic segments, and determined size factors based on the host transcriptome alone.
Genes whose expression correlated with influenza ranking were then determined using the Monocle function differentialGeneTest, again removing variance explained by sample to sample variation and using a q-value threshold of 0.1. 
Cells were then arranged my ranking within each sample, in order of increasing percentage of reads derived from influenza within a sample, and differentially-regulated genes were visualized.


\section{Acknowledgments}

Additional information can be given in the template, such as to not include funder information in the acknowledgments section.

\nocite{*} % This command displays all refs in the bib file
\bibliography{references}


\end{document}


\end{document}